### PR TITLE
SAMZA-1550: Doc for 0.14.0 release

### PR DESCRIPTION
**Pull request by xinyuiscool**
Docs update for both master and 0.14.0 branch.

### CODE REVIEW

File: docs/startup/releases/versioned/release-notes.md [hunk 1]
```diff
@@ -42,8 +43,12 @@ title: Release Notes
 
 ## Release Notes
 <!-- Add notes on new features, modified behavior of existing features, operational/performance improvements, new tools etc -->
-
-<TBD>
+* [SAMZA-1510](https://issues.apache.org/jira/browse/SAMZA-1510) - Samza SQL
+* [SAMZA-1438](https://issues.apache.org/jira/browse/SAMZA-1438) - Event-hub connectors for Samza
```
Comment (nickpan47) [inline docs/startup/releases/versioned/release-notes.md]: I thought that KinesisSystemConsumer is also included in 0.14?

Comment (nickpan47) [reply docs/startup/releases/versioned/release-notes.md]: SAMZA-1515 is the ticket for Kinesis system consumer.

Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: s/Event-hub connectors for Samza /Producer and consumer for Azure EventHubs[link]"

File: docs/startup/releases/versioned/release-notes.md [hunk 2]
```diff
@@ -42,8 +43,12 @@ title: Release Notes
...
+* [SAMZA-1421](https://issues.apache.org/jira/browse/SAMZA-1421) - Add support for durable state for high level API
```
Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: 1.s/high level/high-level
2.for high level API -> in high-level API
3.s/Add support for/Support for ..

File: docs/startup/releases/versioned/release-notes.md [hunk 3]
```diff
@@ -42,8 +43,12 @@ title: Release Notes
...
+* [SAMZA-1392](https://issues.apache.org/jira/browse/SAMZA-1392) - KafkaSystemProducer performance and correctness with concurrent sends and flushes
```
Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: s/KafkaSystemProducer performance and correctness with concurrent sends and flushes/"Performance and correctness fixes for KafkaSystemProducer for concurrent sends and flushes"

File: docs/startup/releases/versioned/release-notes.md [hunk 4]
```diff
@@ -42,8 +43,12 @@ title: Release Notes
...
+* [SAMZA-1406](https://issues.apache.org/jira/browse/SAMZA-1406) - Enhancements to the ZooKeeper based deployment model
```
Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: s/Zookeeper based/Zookeeper-based

File: docs/startup/releases/versioned/release-notes.md [hunk 5]
```diff
@@ -42,8 +43,12 @@ title: Release Notes
...
+* [SAMZA-1321](https://issues.apache.org/jira/browse/SAMZA-1321) - Add support for multi-stage batch processing
```
Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: s/Add support for../Support for

File: docs/startup/releases/versioned/release-notes.md (hunk 1 above)
Comment (vjagadish1989) [inline docs/startup/releases/versioned/release-notes.md]: Can we also include : 

Checkpoint provider for Azure table? (SAMZA-1486)

**Comment by vjagadish1989** [comment]
Mostly minor feedback on technical documentation.

File: docs/startup/releases/versioned/release-notes.md (hunk 1 above)
Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: Added.

Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: Fixed.

Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: Added.

File: docs/startup/releases/versioned/release-notes.md (hunk 2 above)
Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: Fixed.

File: docs/startup/releases/versioned/release-notes.md (hunk 4 above)
Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: fixed.

File: docs/startup/releases/versioned/release-notes.md (hunk 5 above)
Comment (xinyuiscool) [reply docs/startup/releases/versioned/release-notes.md]: fixed.

**Pull request closed without merging**